ocs_ci/deployment/cnv.py

```diff
@@ -0,0 +1,505 @@
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.resources.packagemanifest import PackageManifest
+from ocs_ci.ocs.constants import (
+    CNV_NAMESPACE_YAML,
+    CNV_OPERATORGROUP_YAML,
+    CNV_SUBSCRIPTION_YAML,
+    CNV_CATALOG_SOURCE_YAML,
+    CNV_HYPERCONVERGED_YAML,
+    CNV_OPERATOR_NIGHTLY_CATALOG_SOURCE_NAME,
+)
+from ocs_ci.utility import templating
+from ocs_ci.ocs import constants
+from ocs_ci.utility.utils import run_cmd
+from ocs_ci.ocs import exceptions
+from ocs_ci.ocs.resources.catalog_source import CatalogSource
+from ocs_ci.ocs.resources.install_plan import wait_for_install_plan_and_approve
+from ocs_ci.ocs.resources.csv import CSV, get_csvs_start_with_prefix
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.ocs import ocp
+from ocs_ci.ocs.resources.pod import wait_for_pods_to_be_running
+from ocs_ci.ocs.node import wait_for_nodes_status
+
+logger = logging.getLogger(__name__)
+
+
+class CNVInstaller(object):
+    """
+    CNV Installer class for CNV deployment
+    """
+
+    def __init__(self):
+        self.namespace = constants.CNV_NAMESPACE
+
+    def create_cnv_catalog_source(self):
+        """
+        Creates a nightly catalogsource manifest for CNV operator deployment from quay registry.
+
+        """
+        logger.info("Adding CatalogSource for CNV")
+        cnv_catalog_source_data = templating.load_yaml(CNV_CATALOG_SOURCE_YAML)
+        cnv_image_tag = config.DEPLOYMENT.get("ocs_csv_channel")[-4:]
+        cnv_catalog_source_data["spec"][
+            "image"
+        ] = f"{constants.CNV_QUAY_NIGHTLY_IMAGE}:{cnv_image_tag}"
+        cnv_catalog_source_manifest = tempfile.NamedTemporaryFile(
+            mode="w+", prefix="cnv_catalog_source_manifest", delete=False
+        )
+        templating.dump_data_to_temp_yaml(
+            cnv_catalog_source_data, cnv_catalog_source_manifest.name
+        )
+        run_cmd(f"oc apply -f {cnv_catalog_source_manifest.name}", timeout=2400)
+        cnv_catalog_source = CatalogSource(
+            resource_name=CNV_OPERATOR_NIGHTLY_CATALOG_SOURCE_NAME,
+            namespace=constants.MARKETPLACE_NAMESPACE,
+        )
+
+        # Wait for catalog source is ready
+        cnv_catalog_source.wait_for_state("READY")
+
+    def create_cnv_namespace(self):
+        """
+        Creates the namespace for CNV resources
+
+        Raises:
+            CommandFailed: If the 'oc create' command fails.
+
+        """
+        try:
+            logger.info(f"Creating namespace {self.namespace} for CNV resources")
+            namespace_yaml_file = templating.load_yaml(CNV_NAMESPACE_YAML)
+            namespace_yaml = OCS(**namespace_yaml_file)
+            namespace_yaml.create()
+            logger.info(f"CNV namespace {self.namespace} was created successfully")
+        except exceptions.CommandFailed as ef:
+            if (
+                f'project.project.openshift.io "{self.namespace}" already exists'
+                not in str(ef)
```

Comment:
isn't this supposed to be
```suggestion
          if (f'project.project.openshift.io "{self.namespace}" already exists' in str(ef))
```
??

Reply:
Done